### PR TITLE
Improvements to the NASM samples 

### PR DESCRIPTION
**Pull request by bernardosulzbach**
When reviewing #483, I set out to test how complete the NASM support was.

Apparently, an issue with brackets has already been resolved.

Nonetheless, I went ahead and performed some consistency refactorings in the NASM samples (it mixed tabs with spaces, had misaligned comments, missing spaces after commas, and so on). I also added compilation instructions to the lengthier sample as it is interesting to be able to check whether or not it is a working program.

### CODE REVIEW

File: spec/visual/samples/nasm [hunk 1]
```diff
@@ -1,9 +1,15 @@
-;Simple test of NASM parser
+; Simple test of the NASM parser
+;
+; Sample compilation (requires 32-bit support from NASM and GCC).
```
Comment (pyrmont) [inline spec/visual/samples/nasm]: I'm not sure I understand what this adds. The purpose of the visual sample is to check the lexer is working correctly. There is already sufficient checking of the commenting syntax.

Comment (bernardosulzbach) [reply spec/visual/samples/nasm]: I understand you. However, I'd argue most on a Ruby project do not know NASM syntax or how to test if a NASM file is a valid program that is working as expected. Those two lines are enough to test if the contents of the example are syntactically valid and semantically sound.

However, I am willing to remove these extra lines if they seem excessive to you.

Comment (pyrmont) [reply spec/visual/samples/nasm]: That's true but it's also true of most languages for which we have lexers. It's still the case that we have many visual samples that don't meet this criterion so I apologise if it seems arbitrary but ideally I am trying to move visual samples to be more targeted examples that contain less duplication. If you could remove these lines, that'd be much appreciated.

Comment (bernardosulzbach) [reply spec/visual/samples/nasm]: > (...) if it seems arbitrary (...)

It does not, and your point makes sense as the other visual samples really do not have instructions.

I have removed these lines.

Comment (pyrmont) [reply spec/visual/samples/nasm]: Thanks!

**Pull request merged**